load assets browser before fetch completes and show loading state

christian-byrne opened the pull request:
## Summary

Moves the fetch and post-fetch logic associated with the asset browser into the component and shows a loading state while fetching.

To test, use this branch: https://github.com/comfyanonymous/ComfyUI/pull/10045


https://github.com/user-attachments/assets/718974d5-efc7-46a0-bcd6-e82596d4c389

┆Issue is synchronized with this [Notion page](https://www.notion.so/PR-6189-load-assets-browser-before-fetch-completes-and-show-loading-state-2946d73d365081879d1bd05d86e8c036) by [Unito](https://www.unito.io)

christian-byrne committed:
use async state on asset browser

christian-byrne committed:
fix tag shouldn't be translated

arjansingh reviewed (dismissed):
One nit (not blocking) otherwise we're good.

arjansingh commented on src/platform/assets/components/AssetBrowserModal.vue:
```diff
@@ -65,29 +71,86 @@ const props = defineProps<{
+    await execute()
+  },
+  { immediate: true }
+)
```
I see you @DrJKL!

DrJKL replied:
😜

christian-byrne committed:
[fix] update vue-i18n mock to handle translation params in AssetBrowserModal test

The vue-i18n mock's t function was not accepting params, causing the
displayTitle test to fail. Updated mock to match the @/i18n mock pattern
which correctly handles params and formats them for assertions.

DrJKL commented on src/platform/assets/schemas/assetSchema.ts:
```diff
@@ -4,9 +4,9 @@ import { z } from 'zod'
   id: z.string(),
   name: z.string(),
-  asset_hash: z.string().optional(),
+  asset_hash: z.string().nullish(),
```
Changing something from optional to required but nullable can be risky.

christian-byrne replied:
On the OSS, it returns null, on the cloud, it's undefined. Thanks for flagging this out - cc @arjansingh

DrJKL commented on tests-ui/platform/assets/components/AssetBrowserModal.test.ts:
```diff
@@ -1,32 +1,40 @@
 import type { AssetItem } from '@/platform/assets/schemas/assetSchema'
 
-// Mock @/i18n for useAssetBrowser and AssetFilterBar
+const mockAssetService = vi.hoisted(() => ({
```
Ooooh, very nice.

christian-byrne committed:
[fix] remove formattedSize test after feature was removed in main

The formattedSize feature was intentionally removed in commit 4555656bf
(#6227) because it was confusing users. During rebase conflict resolution,
the test for this feature was accidentally kept. Removing it now.

actions-user committed:
[automated] Apply ESLint and Prettier fixes

christian-byrne commented on src/platform/assets/composables/useAssetBrowser.ts:
```diff
@@ -67,7 +68,10 @@ export interface AssetDisplayItem extends AssetItem {
  */
-export function useAssetBrowser(assets: AssetItem[] = []) {
+export function useAssetBrowser(
+  assetsSource: Ref<AssetItem[] | undefined> = ref<AssetItem[] | undefined>([])
```
> make composables more flexible by using MaybeRefOrGetter and toValue

YAGNI
